File: pytools/data/data.code.golang.client/ack_test_x_x.go
package proto

import (
	"packet"
)

type AckTestXX struct {
	idU8                     uint8
	idU16                    uint16
	idU32                    uint32
	repeatIdU8               []uint8
	optionalIdU8Flag         uint8
	optionalIdU8             uint8
}

func AckTestXXDecode(pack *packet.Packet) *AckTestXX {
	ackTestXX := &AckTestXX{}

	ackTestXX.idU8 = pack.ReadUint8()
	ackTestXX.idU16 = pack.ReadUint16()
	ackTestXX.idU32 = pack.ReadUint32()
	repeatIdU8Count := pack.ReadUint16()
	for ;repeatIdU8Count > 0; repeatIdU8Count-- {
		ackTestXX.repeatIdU8 = append(ackTestXX.repeatIdU8, pack.ReadUint8())
	}
	ackTestXX.optionalIdU8Flag = pack.ReadUint8()
	if ackTestXX.optionalIdU8Flag == 1 {
		ackTestXX.optionalIdU8 = pack.ReadUint8()
	}
	return ackTestXX
}

func (this *AckTestXX) GetIdU8() uint8 {
	return this.idU8
}

func (this *AckTestXX) GetIdU16() uint16 {
	return this.idU16
}

func (this *AckTestXX) GetIdU32() uint32 {
	return this.idU32
}

func (this *AckTestXX) GetRepeatIdU8() []uint8 {
	return this.repeatIdU8
}

func (this *AckTestXX) GetOptionalIdU8() uint8 {
	return this.optionalIdU8
}

File: pytools/data/data.code.lua.client/AckTestXX.lua
module("AckTestXX", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckTestXX})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.id_u8 = 0
	self.id_u16 = 0
	self.id_u32 = 0
	self.repeat_id_u8 = {}
	self.optional_id_u8_flag = 0
	self.optional_id_u8 = 0
end


function decode(self, pack)
	self.id_u8 = pack:readU8()
	self.id_u16 = pack:readU16()
	self.id_u32 = pack:readU32()
	repeat_id_u8_count = pack:readU16()
	for i = 1, repeat_id_u8_count do
		table.insert(self.repeat_id_u8, pack:readU8())
	end
	self.optional_id_u8_flag = pack:readU8()
	if self.optional_id_u8_flag then
		self.optional_id_u8 = pack:readU8()
	end
	return self
end


function getIdU8(self)
	return self.id_u8
end

function getIdU16(self)
	return self.id_u16
end

function getIdU32(self)
	return self.id_u32
end

function getRepeatIdU8(self)
	return self.repeat_id_u8
end

function getOptionalIdU8(self)
	return self.optional_id_u8
end

